packages/mui-utils/src/usePreviousProps.ts

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react';
 
-const usePreviousProps = (value: object): object => {
-  const ref = React.useRef({});
+type AddUndefined<Value> = {
```

Comment:
Added undefined to all keys because initial values of these keys will be undefined